source/ConversationPanel.h

```diff
@@ -124,6 +125,10 @@ template <class T>
 	// If specified, this is a star system to display with a special big pointer
 	// when the player brings up the map. (Typically a mission destination.)
 	const System *system;
+	// If specified, this is a pointer to the ship that this conversation
+	// acts upon (e.g. the ship failing a "flight check", or the NPC you
+	// have boarded).
+	std::shared_ptr<Ship> ship = nullptr;
```

Comment:
shared_ptr<> doesn't need to be initialized, because its default constructor will set it to null.